src/foam/mlang/mlang.js

```diff
@@ -3225,6 +3225,49 @@ foam.CLASS({
 
   documentation: 'A Binary Predicate which applies arg2.f() to arg1.f().',
 
+  properties: [
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg1'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg2'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      code: function(o) {
+        return this.arg2.f(this.arg1.f(o));
+      },
+      javaCode: `
+        Object e1 = getArg1().f(obj);
+        if ( e1 == null ) return e1;
```

Comment:
the reason for this change is that ref could return null

Reply:
rename e1 to receiver and return null instead of the variable.

Reply:
fixed